src/WebJobs.Script.WebHost/Middleware/HstsConfigurationMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.HttpsPolicy;
+using Microsoft.Azure.WebJobs.Script.WebHost.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class HstsConfigurationMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IScriptWebHostEnvironment _webHostEnvironment;
+        private HstsMiddleware _hstsMiddleware;
+        private HostHstsOptions _hostHstsOptions;
+
+        public HstsConfigurationMiddleware(RequestDelegate next, IScriptWebHostEnvironment webHostEnvironment)
+        {
+            _next = next;
+            _webHostEnvironment = webHostEnvironment;
+        }
+
+        public async Task Invoke(HttpContext httpContext, IOptions<HostHstsOptions> options)
+        {
+            if (_hostHstsOptions != options.Value)
+            {
+                _hostHstsOptions = options.Value;
+                _hstsMiddleware = new HstsMiddleware(_next, options);
+            }
+
+            if (!_webHostEnvironment.InStandbyMode && _hostHstsOptions.IsEnabled)
```

Comment:
Why don't we want this for standby mode?

Reply:
This being a new feature most of customers would not have this configuration, So during specialization we would have to create a new `HstsMiddleware` regardless. Even for the customers who would have the default configuration (`isEnabled` set tot true) a new instance of `HostHstsOptions` would be created since the host.json is read again. That would lead to creation of new `HstsMiddleware` even when we provision one during standby mode.

Reply:
Why would we have this for the placeholder? The default is to have this disabled, it would make sense to match that behavior there. The one argument is to make sure that code path is exercised.

Reply:
The other argument is that, in case of issues with the specialization, this feature would be enabled by default. Based on the nature of the change, we wouldn't want the opposite to happen.